### PR TITLE
add table properties for br online restore

### PR DESCRIPTION
**Pull request by lhy1024**
###  What have you changed?

Please explain in detail what the changes are in this PR and why they are needed:

- Create sst writer before download to set table properties.

Don't assume reviewers understand the original issue.

###  What is the type of the changes?

Pick one of the following and delete the others:

- New feature (a change which adds functionality)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test

### CODE REVIEW

**Comment by lhy1024** [comment]
@overvenus @Connor1996 PTAL

File: components/engine_rocks/src/table_properties.rs [hunk 1]
```diff
@@ -31,6 +31,10 @@ impl RocksTablePropertiesCollection {
     pub fn from_raw(raw: RawTablePropertiesCollection) -> RocksTablePropertiesCollection {
         RocksTablePropertiesCollection(raw)
     }
+
+    pub fn get_raw(&self) ->&RawTablePropertiesCollection{
```
Comment (kennytm) [inline components/engine_rocks/src/table_properties.rs]:
```suggestion
    pub fn get_raw(&self) -> &RawTablePropertiesCollection{
```

File: components/sst_importer/src/sst_importer.rs [hunk 2]
```diff
@@ -855,9 +861,9 @@ mod tests {
         // performs the download.
         let importer_dir = tempfile::tempdir().unwrap();
         let importer = SSTImporter::new(&importer_dir).unwrap();
-
+        let sst_writer = <TestEngine as SstExt>::SstWriterBuilder::new().build(importer.get_path(&meta).to_str().unwrap()).unwrap();
```
Comment (kennytm) [inline components/sst_importer/src/sst_importer.rs]: This is too complex to be repeated so many times. Can you please put this into a function like `create_sample_external_sst_file_txn_default()`?

File: src/import/sst_service.rs [hunk 3]
```diff
@@ -164,6 +164,8 @@ impl<Router: RaftStoreRouter> ImportSst for ImportSSTService<Router> {
         let timer = Instant::now_coarse();
         let importer = Arc::clone(&self.importer);
         let limiter = self.limiter.clone();
+        let engine = Arc::clone(&self.engine);
+        let sst_writer =<RocksEngine as SstExt>::SstWriterBuilder::new().set_db(RocksEngine::from_ref(&engine)).build(self.importer.get_path(req.get_sst()).to_str().unwrap()).unwrap();
```
Comment (kennytm) [inline src/import/sst_service.rs]:
```suggestion
        let sst_writer = <RocksEngine as SstExt>::SstWriterBuilder::new().set_db(RocksEngine::from_ref(&engine)).build(self.importer.get_path(req.get_sst()).to_str().unwrap()).unwrap();
```

Consider applying `rustfmt` on this statement

**Comment by lhy1024** [comment]
@overvenus @kennytm @Connor1996 PTAL

**Pull request merged**